scripts/src/public/services/orders.ts

```diff
@@ -63,103 +69,114 @@ function orderDbToApi(order: db.Order, logger: LoggerInstance): Order {
 	};
 }
 
-export async function createOrder(
-	offerId: string, userId: string, logger: LoggerInstance): Promise<OpenOrder> {
-	// offer cap logic
+export async function createOrder(offerId: string, userId: string, logger: LoggerInstance): Promise<OpenOrder> {
+	const offer = await offerDb.Offer.findOneById(offerId);
+	if (!offer) {
+		throw new Error(`cannot create order, offer ${ offerId } not found`);
+	}
+
+	let order = await db.Order.findOne({ where: {
+			userId,
+			offerId,
+			status: "opened"
+		}
+	});
 
-	const openOrder = new db.OpenOrder(offerId, userId);
+	if (!order) {
+		const fn = async () => {
+			const total = await db.Order.count({
+				where: {
+					offerId
+				}
+			});
+
+			if (total === offer.cap.total) {
+				return undefined;
+			}
+
+			const forUser = await db.Order.count({
+				where: {
+					userId,
+					offerId
+				}
+			});
+
+			if (forUser === offer.cap.per_user) {
+				return undefined;
+			}
```

Comment:
This logic can be shared with getOffers

Reply:
plus, in getOffers you return `null` and here you return `undefined`

Reply:
in both cases what i wrote is temporary, we'll need to change it as it's not optimal AT ALL.